core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -183,6 +185,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     val updateMetadataResponse =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         replicaManager.maybeUpdateMetadataCache(correlationId, updateMetadataRequest, metadataCache)
+        if(controller.isActive && !updateMetadataRequest.partitionStates.isEmpty) {
+          updateMetadataRequest.partitionStates.asScala.keys.map(_.topic).toSet { topic =>
+            adminManager.tryCompleteDelayedTopicOperations(topic)
```

Comment:
This one has side effect. So, we should call foreach on the set to make it clear.